7 final

kolcman opened the pull request.

kolcman committed:
Закончил основную логику приложения.

kolcman committed:
Закончил основную логику приложения.

purpleschoolofficial commented on src/App.vue:
```diff
@@ -1,11 +1,15 @@
-        </ul>
+        <Button v-if="!isGameStarted" class="btn btn-start" aria-label="Начать новую игру" @click="startGame()">Начать
+            игру</Button>
+        <div v-else class="game">
+            <ul class="card-list">
```
В компоненте App.vue вы используете события flip-card и change-status, однако соответствующие методы flipCard и changeStatus не определены в коде. Это приведет к ошибкам при попытке вызвать эти методы. Необходимо реализовать эти методы для корректной работы приложения.

purpleschoolofficial commented on src/App.vue:
```diff
@@ -14,40 +18,43 @@
 // import Button from '@/components/Button.vue';
 import Header from '@/components/Header.vue';
 import Card from '@/components/Card.vue'
```
Переменная isGameStarted определена как let, хотя должна быть const, так как вы работаете с реактивным ref-объектом. Лучше использовать const для реактивных переменных, чтобы избежать случайного переопределения.

purpleschoolofficial commented on src/App.vue:
```diff
@@ -14,40 +18,43 @@
+        data.value = raw.map((word, index) => ({
+            id: index,
+            word: word.word,
+            translation: word.translation,
+            state: false,
+            status: 'pending',
+        }));
```
В функции restartGame вы сбрасываете очки и загружаете новые данные, но не сбрасываете состояние isGameStarted. Это может привести к некорректному поведению при перезапуске игры. Рекомендуется явно сбрасывать isGameStarted.value = false или true в зависимости от желаемого поведения.

kolcman committed:
Исправления

- Try Catch при запросе к серверу
- Добавлена реактивная переменная ошибки.
- Кнопка заново сбрасывает состояние игры.

kolcman committed:
Добавлены:

- Добавлены номера для карточек.
- Исправлен слушатель на переворот карточки.
